### PR TITLE
Support for symlinks

### PR DESCRIPTION
**Pull request by piranna**
This fixes issue #4

### CODE REVIEW

**Comment by shinnn** [comment]
Thanks, @piranna. Would you mind adding a test for this feature?

1. Add `test-symlink.tar` to https://github.com/kevva/decompress-tar/tree/master/test/fixtures
2. Check if a symlink is actually extracted as it is.

**Comment by piranna** [comment]
Test for symlinks added ;-)

**Comment by shinnn** [comment]
Great work!

I'll merge this if symlink extraction works correctly in [the real use case](https://github.com/kevva/decompress-tar/issues/4#issue-124009801). Can you confirm it?

File: test/test.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ test('decompress a TAR file', function (t) {
 test('strip path level using the `strip` option', function (t) {
 	t.plan(3);
 
-	var file = vinylFile.readSync(path.join(__dirname, 'fixtures/test.tar'));
+	var file = vinylFile.readSync(path.join(__dirname, 'fixtures/test-nested.tar'));
```
Comment (shinnn) [inline test/test.js]: :+1:

Comment (piranna) [inline test/test.js]: yeah, it was strange to me that nobody was using that tar file... :-P

**Comment by piranna** [comment]
> Great work! I'll merge this if symlink extraction works correctly in the real use case.
> 
> @piranna Can you confirm it?

Good news: I can confirm it :-D
Bad news: our princess is in another castle :-P

At first it was not working, but also by adding some traces they were not printed. The problem was that Linux kernel source code is not in tar format but instead tar.gz, so the fix should have been done on the decompress-targz module :-P Doing the fix there it works and the symlink on Linux source code is correctly creatred, but I've review both projects and are the same, only that decompress-targz calls first to unzip, just a difference of 4 lines. I think both of them could be unified on a single one by using [gunzip-maybe](https://github.com/mafintosh/gunzip-maybe) as tar-stream recommends, so this way there's no duplicated code. What do you think?

**Comment by shinnn** [comment]
Merging this, because we should fix the problem that users currently running into, although there is room for discussion about plugin unification.

**Comment by piranna** [comment]
Ok, thank you. Should I open a new issue for the plugin unification?

**Comment by shinnn** [comment]
> Should I open a new issue for the plugin unification?

Please :)

**Comment by shinnn** [comment]
Planning to publish it as 	v3.1.1. I think this change is not a new feature but just a bug fix.

**Comment by shinnn** [comment]
Ah, I forgot I don't have access to the `decompress-tar` npm package.

@kevva Can you publish v3.1.1?

**Pull request merged**